overlay.nix

```diff
@@ -142,6 +117,14 @@ in
           }
           echo Clients done.
         '';
+
+      writeBashBinChecked = name: final.writeBashChecked "/bin/${name}";
+      writeBashChecked = final.writers.makeScriptWriter {
+        interpreter = "${final.bash}/bin/bash";
+        check = final.writers.writeBash "shellcheck-check" ''
+          ${final.shellcheck}/bin/shellcheck "$1"
+        '';
+      };
```

Comment:
I think these can be replaced with `writeShellApplication` now?

Reply:
Agreed to do this in a future PR.